test/e2e/packetcapture_test.go

```diff
@@ -0,0 +1,728 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	agentconfig "antrea.io/antrea/pkg/config/agent"
+	"antrea.io/antrea/pkg/features"
+)
+
+var (
+	pcSecretNamespace = "kube-system"
+	// #nosec G101
+	pcSecretName     = "antrea-packetcapture-fileserver-auth"
+	tcpServerPodName = "tcp-server"
+	pcToolboxPodName = "toolbox"
+	udpServerPodName = "udp-server"
+	nonExistPodName  = "non-existing-pod"
+
+	tcpProto  = intstr.FromString("TCP")
+	icmpProto = intstr.FromString("ICMP")
+	udpProto  = intstr.FromString("UDP")
+
+	testServerPort   int32 = 80
+	testNonExistPort int32 = 8085
+
+	pcTimeoutReason = "PacketCapture timeout"
+	pcShortTimeout  = uint16(5)
+)
+
+type pcTestCase struct {
+	name           string
+	pc             *crdv1alpha1.PacketCapture
+	expectedStatus crdv1alpha1.PacketCaptureStatus
+
+	// required IP version, skip if not match.
+	ipVersion int
+	// Source Pod to run ping for live-traffic PacketCapture.
+	srcPod string
+}
+
+func genSFTPService() *v1.Service {
```

Comment:
this can be left as future work, but is there no unification possible with the supportbundle e2e test for the VM agent?

Reply:
ok.